Assets/Mirror/Components/NetworkLobbyPlayer.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace Mirror
+{
+    [DisallowMultipleComponent]
+    [AddComponentMenu("Network/NetworkLobbyPlayer")]
+    [HelpURL("https://vis2k.github.io/Mirror/Components/NetworkLobbyPlayer")]
+    public class NetworkLobbyPlayer : NetworkBehaviour
+    {
+        public bool ShowLobbyGUI = true;
+
+        [SyncVar]
+        public bool ReadyToBegin = false;
+
+        [SyncVar]
+        public int Index;
+
+        /// <summary>
+        /// Do not use Start - Override OnStartrHost / OnStartClient instead!
+        /// </summary>
+        public void Start()
+        {
+            if (isClient) SceneManager.sceneLoaded += ClientLoadedScene;
+
+            if (NetworkManager.singleton as NetworkLobbyManager)
+            {
+                ReadyToBegin = false;
+                OnClientEnterLobby();
+            }
+            else
+                Debug.LogError("LobbyPlayer could not find a NetworkLobbyManager. The LobbyPlayer requires a NetworkLobbyManager object to function. Make sure that there is one in the scene.");
+        }
+
+        void OnDisable()
+        {
+            SceneManager.sceneLoaded -= ClientLoadedScene;
+        }
+
+        void ClientLoadedScene(Scene arg0, LoadSceneMode arg1)
+        {
+            NetworkLobbyManager lobby = NetworkManager.singleton as NetworkLobbyManager;
+            if (lobby)
+            {
+                // dont even try this in the startup scene
+                if (SceneManager.GetActiveScene().name == lobby.LobbyScene)
+                {
+                    //Application.targetFrameRate = 10;
+                    return;
+                }
+                //else
+                //    Application.targetFrameRate = 60;
+            }
+
+            if (this != null && isLocalPlayer)
+                CmdSendLevelLoaded();
+        }
+
+        [Command]
+        public void CmdChangeReadyState(bool ReadyState)
+        {
+            ReadyToBegin = ReadyState;
+            NetworkLobbyManager lobby = NetworkManager.singleton as NetworkLobbyManager;
+            if (lobby)
+                lobby.ReadyStatusChanged();
+        }
+
+        [Command]
+        public void CmdSendLevelLoaded()
+        {
+            NetworkLobbyManager lobby = NetworkManager.singleton as NetworkLobbyManager;
+            if (lobby)
+                lobby.PlayerLoadedScene(GetComponent<NetworkIdentity>().connectionToClient);
+        }
+
+        #region lobby client virtuals
+
+        public virtual void OnClientEnterLobby() { }
+
+        public virtual void OnClientExitLobby() { }
+
+        public virtual void OnClientReady(bool readyState) { }
+
+        #endregion
+
+        #region optional UI
+
+        public virtual void OnGUI()
```

Comment:
Perhaps this could be in a different class?  such as NetworkLobbyPlayerGUI  instead of using the ShowLobbyGUI boolen

Reply:
It's a bare bones default.  Users are expected to turn it off and design the scene with their own canvas.